tutorials/getting-started/custom-widgets.md

```diff
@@ -0,0 +1,246 @@
+---
+title: "Custom widgets"
+---
+
+## Building custom widgets
+
+You've seen a lot of the widgets that come "in the box" with Apostrophe. But you can do much more by creating your own.
+
+### Custom navigation
+
+A common case: you want to build your own navigation menu. [Apostrophe's page tree is awesome](building-navigation.html) but sometimes you want to "cherrypick" pages from all over the tree, perhaps for a special footer.
+
+You could use a rich text widget and just tell users to add links manually. But they'll just break each time a page is moved around the site. And it's easier for users to mess up the formatting that way. We want something more consistent.
+
+Let's look at some custom widgets that help provide navigation. We'll start with a simple widget that adds a link in a well-formatted way.
+
+#### `link`: the simplest widget
+
+First we'll need a folder for the module:
+
+```bash
+mkdir -p lib/modules/link-widgets
+```
+
+> Just about everything new you create in Apostrophe will be a "module." Project-specific modules live in `lib/modules`. Apostrophe will spot them there if you list them in `app.js`.
+> You can also publish and install modules with `npm`. Apostrophe looks in both places. Your module name **should not start with `apostrophe`**. That's reserved for our official modules.
+> Modules almost always have plural names. The name of a module that provides widgets should end in `-widgets`.
+
+Now we'll write the code for our module in `lib/modules/custom-navigation-widgets/index.js`:
+
+```javascript
+module.exports = {
+  extend: 'apostrophe-widgets',
+  label: 'Link to Anywhere',
+  addFields: [
+    {
+      name: 'url',
+      type: 'url',
+      label: 'URL',
+      required: true
+    },
+    {
+      name: 'label',
+      type: 'string',
+      label: 'Label',
+      required: true
+    }
+  ]
+};
+```
+
+> *"What does `extend` mean here?"* Our module is extending the `apostrophe-widgets` module, which provides almost all the code we need.  Yes, `extend` is the correct spelling. Apostrophe uses [moog](https://npmjs.org/modules/moog) to handle extending or "subclassing" other modules.
```

Comment:
Markdown compiles it to a blockquote, which seems good to me.

On Tue, Jul 5, 2016 at 10:26 AM, Tom Boutell <tom@punkave.com> wrote:

> I am proposing its adoption across our docs, yes. It's a common convention
> in programming books and websites for "read this if you're
> interested/trying to learn a little more," and it's hard to come back to
> this material out of context later.
>
> On Tue, Jul 5, 2016 at 10:04 AM, Jimmy Harrison <notifications@github.com>
> wrote:
>
>> In tutorials/getting-started/custom-widgets.md
>> <https://github.com/punkave/apostrophe-documentation/pull/15#discussion_r69568171>
>> :
>>
>> > +      name: 'url',
>> > +      type: 'url',
>> > +      label: 'URL',
>> > +      required: true
>> > +    },
>> > +    {
>> > +      name: 'label',
>> > +      type: 'string',
>> > +      label: 'Label',
>> > +      required: true
>> > +    }
>> > +  ]
>> > +};
>> > +```
>> > +
>> > +> *"What does `extend` mean here?"* Our module is extending the `apostrophe-widgets` module, which provides almost all the code we need.  Yes, `extend` is the correct spelling. Apostrophe uses [moog](https://npmjs.org/modules/moog) to handle extending or "subclassing" other modules.
>>
>> Is the Q&A aside an element that we want to adopt across our
>> documentation? I can see where it's helpful, but we want to make sure we
>> use it consistently across all of the docs.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/punkave/apostrophe-documentation/pull/15/files/758d508e28dd26bb77ad4869ad70c4cdd051325c#r69568171>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AAB9fU1ZgJGwn9fTNHdlHB96VQ9BLffEks5qSmR8gaJpZM4I7_d4>
>> .
>>
>
>
>
> --
>
>
> *THOMAS BOUTELL, *SUPPORT LEAD
> P'UNK AVENUE | (215) 755-1330  |  punkave.com
>



-- 


*THOMAS BOUTELL, *SUPPORT LEAD
P'UNK AVENUE | (215) 755-1330  |  punkave.com